modules/auxiliary/admin/http/grafana_auth_bypass.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# standard modules
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+import requests 
+import binascii
+import hashlib
+import logging
+import os
+import re
+
+from metasploit import module
+
+
+metadata = {
+    'name': 'Grafana 2.0 through 5.2.2 authentication bypass for LDAP and OAuth',
+    'description': '''
+        This module generates a remember me cookie for a valid username. Through unpropper seeding 
+        while userdate are requested from LDAP or OAuth it's possible to craft a valid remember me cookie. 
+        This cookie can be used for bypass authentication for everyone knowing a valid username.
+    ''',
+    'authors': [
+        'Rene Riedling',
+        'Sebastian Solnica' #Original Discovered
+    ],
+    'date': '2019-08-14', # set to date of creation
+    'license': 'MSF_LICENSE',
+    'references': [
+        {'type': 'cve', 'ref': '2018-15727'},
+        {'type': 'url', 'ref': 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-15727'},
+        {'type': 'url', 'ref': 'https://grafana.com/blog/2018/08/29/grafana-5.2.3-and-4.6.4-released-with-important-security-fix/'}
+    ],
+    'type': 'single_scanner',
+    'options': {
+        'VERSION': {'type': 'string', 'description': 'Grafana version', 'required': True, 'default': '5'},
```

Comment:
This should be an enum of `'2-4', '5'`, or better yet if you could I would recommend automatically fingerprinting it based on the footer in the login HTML page.

```html
<li>
  <a href="https://grafana.com" target="_blank">Grafana</a>
  <span>v5.2.1 (commit: 2040f61)</span>
</li>
```